Add integration test for mongodb receiver

Frapschen opened the pull request:
**Description:** Add integration test for mongodb receiver

Resolves #10864

Frapschen committed:
init

Frapschen committed:
add integration test

Frapschen committed:
fix error update

Frapschen committed:
Update scraper_test.go

Frapschen committed:
Update ingegration_test.go

djaglowski commented on receiver/mongodbreceiver/ingegration_test.go:
```diff
@@ -0,0 +1,154 @@
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
+
+		expectedFile := filepath.Join("testdata", "integration", "expected.5_0.json")
```
How was this file created?

Typically, it should be written from the actual result. (One time during development, and then the `WriteMetrics` call should be removed.) There should not be an "actual metrics" file, only an "expected metrics" file.

djaglowski commented on receiver/mongodbreceiver/ingegration_test.go:
```diff
@@ -0,0 +1,154 @@
+
+		actualMtrics := consumer.AllMetrics()[0]
+
+		actualFile := filepath.Join("testdata", "integration", "actual.5_0.json")
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
```
This should be considered temporary code.
```suggestion
		actualFile := filepath.Join("testdata", "integration", "expected.5_0.json")
		err = golden.WriteMetrics(actualFile, actualMtrics)
		require.NoError(t, err)
```

Run this one time and then remove all of these lines. Then you have your "expected" file.

Frapschen replied:
ok, I have known how to do next.

djaglowski commented on receiver/mongodbreceiver/ingegration_test.go:
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build integration
```
Please double check the name of this file.

Frapschen replied:
ok

Frapschen committed:
apply reviewer's suggestion:

- Use temporary code to gen "expected" file.
- Run those temporary code one time and then remove all of they.

Frapschen committed:
Merge remote-tracking branch 'origin/Add_integration_test_for_mongodb_receiver' into Add_integration_test_for_mongodb_receiver

Frapschen commented:
@djaglowski review pls :)

Frapschen committed:
Merge branch 'main' into Add_integration_test_for_mongodb_receiver

# Conflicts:
#	CHANGELOG.md

Frapschen committed:
change port for Parallel running.

Frapschen committed:
change import style

djaglowski commented on receiver/mongodbreceiver/integration_test.go:
```diff
@@ -0,0 +1,146 @@
+
+func TestMongodbIntegration(t *testing.T) {
+	t.Run("Running mongodb 4.0", func(t *testing.T) {
+		t.Parallel()
```
```suggestion
```

djaglowski commented on receiver/mongodbreceiver/integration_test.go:
```diff
@@ -0,0 +1,147 @@
+		require.NoError(t, err)
+	})
+	t.Run("Running mongodb 5.0", func(t *testing.T) {
+		t.Parallel()
```
```suggestion
```

Frapschen replied:
I have changed the second test's port,so it can running parallel.

djaglowski reviewed (commented):
```
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.
```

Frapschen committed:
update go.sum

Frapschen committed:
update

djaglowski reviewed (approved):
Thank you @Frapschen!
